File: compiler/src/main/java/com/duy/pascal/backend/exceptions/UnrecognizedTokenException.java
package com.duy.pascal.backend.exceptions;

import com.duy.pascal.backend.tokens.Token;

public class UnrecognizedTokenException extends com.duy.pascal.backend.exceptions.ParsingException {
    public Token token;

    public UnrecognizedTokenException(Token t) {
        super(t.lineInfo, "The following name doesn't belong here: " + t);
        this.token = t;
    }
}

File: libCompiler/src/main/java/com/duy/pascal/backend/exceptions/UnrecognizedTokenException.kt
package com.duy.pascal.backend.exceptions

import com.duy.pascal.backend.tokens.Token

class UnrecognizedTokenException : ParsingException {
    var token: Token

    constructor(token: Token) : super(token.lineNumber, "The following name doesn't belong here: $token") {
        this.token = token
    }
}
